crates/client/src/config.rs

```diff
@@ -0,0 +1,101 @@
+extern crate lazy_static;
+
+use anyhow::Context;
+use fehler::throw;
+use serde::Deserialize;
+use std::{env, fs, io, path::PathBuf};
+use thiserror::Error;
+
+pub const CARGO_TOML: &str = "Cargo.toml";
+pub const TRDELNIK_TOML: &str = "Trdelnik.toml";
+pub const ANCHOR_TOML: &str = "Anchor.toml";
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("invalid workspace")]
+    BadWorkspace,
+    #[error("{0:?}")]
+    Anyhow(#[from] anyhow::Error),
+    #[error("{0:?}")]
+    Io(#[from] io::Error),
+    #[error("{0:?}")]
+    Toml(#[from] toml::de::Error),
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct Test {
+    pub validator_startup_timeout: u64,
+}
+
+#[derive(Default, Debug, Deserialize, Clone)]
+struct _Test {
+    #[serde(default)]
+    pub validator_startup_timeout: Option<u64>,
+}
+
+impl From<_Test> for Test {
+    fn from(_t: _Test) -> Self {
+        Self {
+            validator_startup_timeout: _t.validator_startup_timeout.unwrap_or(10_000),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize, Clone)]
+pub struct Config {
+    pub test: Test,
+}
+
+#[derive(Default, Debug, Deserialize, Clone)]
+struct _Config {
+    #[serde(default)]
+    pub test: Option<_Test>,
+}
+
+impl From<_Config> for Config {
+    fn from(_c: _Config) -> Self {
+        Self {
+            test: _c.test.unwrap_or_default().into(),
+        }
+    }
+}
+
+impl Config {
+    pub fn new() -> Self {
+        let root = Config::discover_root().expect("failed to find the root folder");
+        let s = fs::read_to_string(root.join(TRDELNIK_TOML).as_path())
+            .expect("failed to read the Trdelnik config file");
```

Comment:
I am just thinking if we want this :thinking: The file is required, so it's not backward compatible (which is not a problem). But I would like to ask all of you, what do you think about it? I understand it's a config file, it's just about some collective decision lets say :smile: 

@tribuste @vmarcin

Reply:
This is ok, discussed internally. We will keep it, but please add a line to changelog and readme, that it's not backward compatible and users have to run `trdelnik init` to create a config file otherwise it fail.

Reply:
:heavy_check_mark: I've added a note to the `CAHNGELOG.md`